### DISCUSSION TITLE
How to configure "Traffic forwarding" and "TLS mutual authentication" on a localhost?

### DISCUSSION THREAD

**Discussion by curiosport** [bfenetworks/bfe#929] [Q&A (问题和回答)] [answered]
I am developing everything on the same PC, then when the project goes into production I would simply change the configuration to suit the VPS systems.

I have 1 standalone server listening on port **8000**:
`https://127.0.0.1:8000/`

At the start I don't need anything complex bfe and the load balancer just needs to be ready to add more servers to it in the future.

I followed these tutorials to modify the .conf and .data files:
[Route](https://www.bfe-networks.net/en_us/example/route/)
[TLS client authentication](https://www.bfe-networks.net/en_us/example/client_auth/)

I also went by the example configurations in the source, and well, in short I tried it and I could not make it work, and if it "worked" it was a false positive that I only discovered it when I connected to the `https://127.0.0.1:8443/` address of bfe.

Basically I need help on these configurations:
```
bfe.conf

cluster_conf/
mod_trust_clientip/
server_data_conf/
tls_conf/

```

**Comment by mileszhang2016** [comment, marked as answer]
Please refer to Chapter 19 to Chapter 22 of the following book.
https://github.com/baidu/bfe-book

**Comment by clarinette9** [comment]
Hi curiosport, this is Jian from bfe team. 
Thank you for your contribution to bfe.  If you can,  drop me an email.
Or  joint our Wechat group by scan QR code displayed at bottom of  this page: https://github.com/bfenetworks/bfe/blob/develop/README-CN.md

**Reply by curiosport** [reply]
I sent you an email, my address is ended in @**engineer.com**
